### PR TITLE
Feature/only one calendar option

### PR DESCRIPTION
**Pull request by martinmunillas**
What's in this pr:

- Rewired whole internals, moved all event handlers and useful info to a context, also made `Day` some props that  where javascript `Dates` and could be converted to `Day` s
- Added the posibility of selecting just one date
- Added responsiveness to the component
- Improved Accessibility
- Replaced Styled Components with Emotion/Styled
- Changed github actions so gets only published on realease

### CODE REVIEW

**Review by martinmunillas — COMMENTED** [review]
there might be some breaking changes, but nothing to worry about, just if you check the maxDate and minDate prop type, changed the type

File: package.json [hunk 1]
```diff
@@ -53,9 +52,9 @@
     "typescript": "^4.2.3"
   },
   "dependencies": {
-    "wtw-icons": "^1.13.0",
+    "@emotion/styled": "^11.3.0",
```
Comment (ARivas16) [inline package.json:55]: Just a doubt: what is the difference between this package and styled components?

File: src/utils/hooks/useDateSelector.ts [hunk 2]
```diff
@@ -5,8 +5,11 @@ import { dayIsSooner } from "../funcs/dayIsSooner";
 import { generateButtonId } from "../funcs/generateButtonId";
 import { generateDay } from "../funcs/generateDay";
 
-export const useDateSelector = (initial?: Tuple<Day | null, 2>) => {
-  const [selected, setSelected] = useState<Tuple<Day | null, 2>>(
+export const useDateSelector = (
+  initial?: Tuple<Day | null, 1 | 2>,
+  isMultple = true
```
Comment (ARivas16) [inline src/utils/hooks/useDateSelector.ts:10]: typo `Multiple`

File: src/utils/hooks/useDateSelector.ts [hunk 3]
```diff
@@ -15,23 +18,26 @@ export const useDateSelector = (initial?: Tuple<Day | null, 2>) => {
   );
 
   const addDate = (day: Day) => {
-    if (!selected[0]) return setSelected([day, null]);
+    if (isMultple) {
```
Comment (ARivas16) [inline src/utils/hooks/useDateSelector.ts:21]: same as before

**Review by ARivas16 — CHANGES_REQUESTED** [review]
Just a typo! Good job! The code looks cleaner and readable. I have a doubt, this allows the selection of a single date and either shows a single calendar or it wasn't changed yet?

File: package.json (hunk 1 above)
Comment (ARivas16) [reply package.json:55]: I didn't watch your daily, I understand now the change :+1:

**Comment by ARivas16** [comment]
> Just a typo! Good job! The code looks cleaner and readable. I have a doubt, this allows the selection of a single date and either shows a single calendar or it wasn't changed yet?

I didn't watch your daily, I know that it was taken too

**Review by ARivas16 — APPROVED** [review]

**Pull request merged**